client/add/add.ctrl.js

```diff
@@ -123,11 +128,16 @@ module.exports = class Add {
            toastAction,
            toastAction ? `/pokemon/${successfulUploads[0].created.id}` : undefined);
       })
-      .filter(line => line.success && line.created.box === box.data.id)
-      .then(lines => lines.slice(0, BOX_PAGE_SIZE - (box.data.contents.length % BOX_PAGE_SIZE)))
+      .filter(line => line.success)
       .map(line => line.created)
-      .then(lines => box.data.contents.push(...lines))
-      .tap(box.onscroll)
+      .filter(line => (boxCounts[line.box]++ % BOX_PAGE_SIZE) < BOX_PAGE_SIZE)
```

Comment:
I'm not sure there's a nice way to do what you want with how I'm doing the filtering. I'm also not sure of a way to do the filtering in a nice way without doing what I was doing before, which is incredibly inefficient. How wedded are you to always having it add more pokemon, even if we already have a lot?